drivers/platform/aducm3029/i2c.c

```diff
@@ -244,14 +267,26 @@ int32_t i2c_read(struct i2c_desc *desc,
 	if (ret < 0)
 		return ret;
 
-	trans->bRepeatStart = (stop_bit == 1) ? 0 : 1;
-	trans->pPrologue = 0;
-	trans->nPrologueSize = 0;
+	if (prologue_size != 0) {
+		trans->bRepeatStart = true;
+		trans->pPrologue = prologue_data;
+		trans->nPrologueSize = prologue_size;
+	} else {
+		trans->bRepeatStart = false;
+		trans->pPrologue = NULL;
+		trans->nPrologueSize = 0;
+	}
+
 	trans->pData = data;
 	trans->nDataSize = bytes_number;
 	trans->bReadNotWrite = 1;
-	if (ADI_I2C_SUCCESS != adi_i2c_ReadWrite(i2c_handler, trans, &errors))
-		return -EIO;
+	ret = adi_i2c_ReadWrite(i2c_handler, trans, &errors);
 
-	return SUCCESS;
+	if (prologue_size != 0) {
```

Comment:
add this cleanup on i2c_remove too

Reply:
Should be done [here](https://github.com/analogdevicesinc/no-OS/pull/1006/files#diff-8c18182bade514b4db3b40669c1670e0da5e78446b38f209375c20083431a4b6R175).